addons/website/models/website.py

```diff
@@ -252,7 +252,7 @@ def _handle_create_write(self, vals):
 
     @api.model
     def _handle_favicon(self, vals):
-        if 'favicon' in vals:
+        if 'favicon' in vals and vals['favicon']:
```

Comment:
This is a bit cleaner, otherwise looks good :+1: 

```suggestion
        if vals.get('favicon'):
```